### PR TITLE
LSIF: allow auth via GITHUB_TOKEN from GitHub Actions

### PR DESCRIPTION
**Pull request by chrismwendt**
This allows LSIF upload authentication via the built-in `GITHUB_TOKEN` in GitHub actions, which eliminates one more manual step from uploading LSIF through a GitHub action.

Unfortunately, this also allows uploads from **any** GitHub app installed on the given repository, **even ones with read-only access**. I haven't found a way to determine that the given token is associated with GitHub Actions and not some other random GitHub App. I also haven't found a zero-impact way to determine that the given token has write access to the repository. @sqs Is this acceptable?

### CODE REVIEW

File: internal/extsvc/github/client.go [hunk 1]
```diff
@@ -251,6 +251,23 @@ func (c *Client) listRepositories(ctx context.Context, requestURI string) ([]*Re
...
 }
 
+// ListInstallationRepositories lists repositories on which the authenticated
+// GitHub App has been installed.
+func (c *Client) ListInstallationRepositories(ctx context.Context) ([]*Repository, error) {
+	type response struct {
+		Repositories []restRepository `json:"repositories"`
+	}
+	var resp response
+	if err := c.requestGet(ctx, "", "installation/repositories", &resp); err != nil {
```
Comment (efritz) [inline internal/extsvc/github/client.go]: If the token is an App installation token and not a personal access token what does the response look like?

Comment (chrismwendt) [reply internal/extsvc/github/client.go]: First few lines of the response:

```json
{
  "total_count": 2,
  "repository_selection": "selected",
  "repositories": [
    {
      "id": 202426614,
```

Comment (efritz) [reply internal/extsvc/github/client.go]: I meant the exact opposite of what I asked.

Comment (chrismwendt) [reply internal/extsvc/github/client.go]: For a PAT, you get 403 Forbidden:

```json
{
  "message": "You must authenticate with an installation access token in order to list repositories for an installation.",
  "documentation_url": "https://developer.github.com/v3/apps/installations/#list-repositories"
}
```

**Comment by sqs** [comment]
> Unfortunately, this also allows uploads from any GitHub app installed on the given repository. I haven't found a way to determine that the given token is associated with GitHub Actions and not some other random GitHub App. @sqs Is this acceptable?

Yeah, that is fine to me. What are you concerned about? Anyone who holds a token that lets them perform write operations on the repository can do much worse than upload bad LSIF data to Sourcegraph.

**Comment by chrismwendt** [comment]
> > Unfortunately, this also allows uploads from any GitHub app installed on the given repository. I haven't found a way to determine that the given token is associated with GitHub Actions and not some other random GitHub App. @sqs Is this acceptable?

> Yeah, that is acceptable. What are you concerned about? Anyone who holds a token that lets them perform write operations on the repository can do much worse than upload bad LSIF data to Sourcegraph.

Read-only access is sufficient to pass the auth check in this PR (clarified the description). I believe only GitHub App owners can create installation access tokens, so the set of people with access is likely very small (~1 per GitHub App installed on a repository).

**Comment by sqs** [comment]
@chrismwendt Got it. That seems fine to me.

**Pull request merged**